### PR TITLE
Add realtime metrics support to Thanos deployments

### PR DESCRIPTION
**Pull request by AAkindele**
# Type of PR

- [x] Documentation changes
- [x] Code changes
- [ ] Test changes
- [ ] CI-CD changes
- [ ] GitHub Template changes

## PR Checklist

- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] My code follows the code style of this project.
- [ ] I ran lint checks locally prior to submission.
- [x] Have you checked to ensure there aren't other open Pull Requests for the same update/change?

## Purpose of PR

- Add Thanos to preprod including real-time metrics
- Update the existing dev deployment to include real-time metrics
- Move Thanos sidecar configuration from environment specific configs to base config
- Increased pod memory limit on Azure policy to allow for more memory on some of the Thanos components
  - TODO: [New task](https://github.com/retaildevcrews/wcnp/issues/1045) to investigate resource usage and recommendations for Thanos
- Added instructions for creating private DNS records for Thanos components to communicate cross-cluster
  - This allows the Thanos Observer cluster to fetch real-time metrics from the Observee clusters

## Does this introduce a breaking change

- [ ] YES
- [x] NO

## Validation

- [ ] Unit tests updated and ran successfully
- [x] Update documentation or issue referenced above

## Issues Closed or Referenced

- Closes https://github.com/retaildevcrews/wcnp/issues/488

### CODE REVIEW

File: cluster/cluster-stamp.json [hunk 1]
```diff
@@ -2104,7 +2104,7 @@
                         "value": "1000m"
                     },
                     "memoryLimit": {
-                        "value": "512Mi"
+                        "value": "1024Mi"
```
Comment (kforeverisback) [inline cluster/cluster-stamp.json:2107]: Will this increase Limits for all containers/pods?

**Review by kforeverisback — APPROVED** [review]
LGTM.

[Optional] Query and concern:
- We've changed cluster-baseline azure policy to increase memory limit to 1000mb. It would apply to all resources. Is this the only way? Is there any pod specific exceptions that we can use to add thanos exception?

File: cluster/cluster-stamp.json (hunk 1 above)
Comment (AAkindele) [reply cluster/cluster-stamp.json:2107]: Yep. This policy update would apply to all the pods.

**Comment by kforeverisback** [comment]
> 

Since we did the bug bash and came to a consensus about the mem limit, I prefer to add Thanos specific exception (if that is possible).
It's my preference, but it might be a good idea to discuss it with the team.

**Comment by AAkindele** [comment]
I can add this to the issue currently in triage https://github.com/retaildevcrews/wcnp/issues/1045.

**Pull request merged**